riscv-sbi.md

```diff
@@ -81,57 +106,64 @@ vector is represented as a sequence of unsigned longs whose length equals the
 number of harts in the system divided by the number of bits in an unsigned long,
 rounded up to the next integer.
 
-If hart_mask is NULL, an IPI is sent to every hart in the system.
+If hart_mask is `NULL`, an IPI is sent to every hart in the system.
 
 ```C
 void sbi_clear_ipi(void)
 ```
 Clears the pending IPIs if any. The IPI is cleared only in the hart for which
 this SBI call is invoked.
 
-### Memory Model<a name=#memory-model>
+### Memory Model<a name="memory-model" />
 ```C
 void sbi_remote_fence_i(const unsigned long *hart_mask)
 ```
-Instruct remote harts to execute FENCE.I instruction.
-N.B. hart_mask is as described in sbi_send_ipi.
+Instruct remote harts to execute a `FENCE.I` instruction.
+`hart_mask` is as described in `sbi_send_ipi`, above.
+
 ```C
 void sbi_remote_sfence_vma(const unsigned long *hart_mask,
-                           unsigned long start,
-                           unsigned long size)
+                           const void *start,
+                           size_t size)
 ```
 //TODO
 
 ```C
 void sbi_remote_sfence_vma_asid(const unsigned long *hart_mask,
-                                unsigned long start,
-                                unsigned long size,
-                                unsigned long asid)
+                                const void *start,
+                                size_t size,
+                                asid_t asid)
 ```
 //TODO
 
-### Console
+### Console<a name="Console" />
 ```C
-void sbi_console_putchar(int ch)
+void sbi_console_putbyte(uint8_t ch)
```

Comment:
I think this is fine, as it doesn't break ABI compatibility (the ABI is the SBI number, not the function name).  We should update the Linux and BBL ports to the new name, do you want to submit patches?

Reply:
Oh, sorry, Andrew's right -- I didn't notice the argument change.